providers/sendgrid/src/lib/sendgrid.provider.ts

```diff
@@ -38,11 +73,18 @@ export class SendgridEmailProvider implements IEmailProvider {
           type: attachment.mime,
         };
       }),
-    });
-
-    return {
-      id: response[0]?.headers['x-message-id'],
-      date: response[0]?.headers?.date,
     };
   }
 }
+
+const mapResponse = (statusCode: number) => {
```

Comment:
Error codes mapping referenced from [Sendgrid API Responses](https://docs.sendgrid.com/api-reference/how-to-use-the-sendgrid-v3-api/responses)